### ISSUE TITLE
CRITICAL: MLM training objective is wrong for Esperanto - need semantic similarity

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#57]
## Problem

The current morpheme-aware training script uses Masked Language Modeling (MLM) as the training objective. This is fundamentally wrong for Esperanto because:

1. **MLM teaches grammar, not semantics**: MLM trains the model to predict masked tokens from context, which primarily teaches grammatical patterns (what word forms fit syntactically)

2. **Esperanto grammar is 100% deterministic**: The 16 rules of Esperanto grammar are perfectly regular with no exceptions. We don't need to LEARN grammar - we can compute it programmatically.

3. **Increasing loss proves the problem**: Training loss went from 6.96 → 7.24 → 7.28 (increasing each epoch). This indicates the model is learning the wrong thing.

## Evidence

From training run (2024-12-24):
```
Epoch 1: 6.9607
Epoch 2: 7.2446 (+4.1%)
Epoch 3: 7.2832 (+4.6%)
```

Loss INCREASED consistently - the model was getting WORSE at the wrong task.

## Root Cause

MLM objective:
```python
# This trains the model to predict masked tokens
loss = cross_entropy(model(masked_sentence), original_tokens)
```

This teaches: "Given context, what word FORM fits here?"
But we want: "What is the MEANING of this root/affix?"

## Correct Approach

**Semantic Similarity Training** instead of MLM:

### For Roots
Train on semantic similarity pairs:
- Positive pairs: Sentences with same topic/meaning (from same Wikipedia article)
- Negative pairs: Sentences with different topics

```python
# Correct objective
loss = contrastive_loss(embed(sentence1), embed(sentence2), is_similar)
```

### For Semantic Affixes
Train on composition relationships:
- `hund + ej` should be similar to embeddings for "kennel-like" contexts
- `bon + mal` should be opposite to `bon`

### For Grammatical Affixes
Don't train at all - use deterministic feature vectors.

## Impact

- All trained models are invalid and have been deleted
- Training scripts need to be rewritten with correct objective
- Supersedes Issues #51, #52, #53, #54, #55, #56

## References

- Wiki: [Esperanto-Grammar-Reference.md](../wiki/Esperanto-Grammar-Reference) - Morpheme classification
- Wiki: [Compositional-Embeddings.md](../wiki/Compositional-Embeddings) - Training objectives

## Action Items

1. [ ] Design semantic similarity training objective for roots
2. [ ] Design affix composition training objective
3. [ ] Reuse existing Tatoeba parallel corpus for similarity pairs
4. [ ] Update training script to use correct objectives
5. [ ] Retrain with new approach

**Comment by marctjones** [comment]
Superseded by #65 (Epic: Fundamento-Centered Training). The new training plan uses definitional semantics from the Fundamento rather than MLM.
